src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingProcessor.java

```diff
@@ -123,13 +123,12 @@ private void updateAndSave(final Hearing hearing) {
     }
 
     Hearing assignUuids(Hearing hearing) {
-        // Apply the new case ID
-        final var caseId = UUID.randomUUID().toString();
-        var updatedHearing = hearing.withCaseId(caseId).withHearingId(caseId);
+        var updatedHearing = hearing.withHearingId(UUID.randomUUID().toString());
 
         // We want to retain the LIBRA case no if present
         if (hearing.getCaseNo() == null) {
-            updatedHearing = updatedHearing.withCaseNo(caseId);
+            updatedHearing = updatedHearing.withCaseNo(hearing.getCaseId());
+            hearing = hearing.withCaseNo(hearing.getCaseId());
```

Comment:
why do we do this? We are returning  `updatedHearing`, not `hearing`

Reply:
Good spot, removed